src/main/java/ru/sbt/mipt/oop/events/DoorEventProcessor.java

```diff
@@ -0,0 +1,26 @@
+package ru.sbt.mipt.oop.events;
+
+import ru.sbt.mipt.oop.SmartHome;
+import ru.sbt.mipt.oop.commands.CommandType;
+import ru.sbt.mipt.oop.elements.Door;
+import ru.sbt.mipt.oop.elements.Room;
+
+import static ru.sbt.mipt.oop.events.SensorEventType.DOOR_CLOSED;
+import static ru.sbt.mipt.oop.events.SensorEventType.DOOR_OPEN;
+
+public class DoorEventProcessor implements EventProcessor{
+    public SensorEvent processEvent(SmartHome smartHome, SensorEvent event) {
+        for (Room room : smartHome.getRooms()) {
+            Door door = room.getDoor(event.getObjectId());
+            if (door != null) {
+                room.changeDoor(door.getId(), (event.getType() == DOOR_OPEN));
+                if (room.getName().equals("hall") && event.getType() == DOOR_CLOSED) {
```

Comment:
https://github.com/iagoshkov/smart-home-2020/blob/master/STUDENT_GUIDE.md#%D1%80%D0%B0%D0%B7%D0%B4%D0%B5%D0%BB%D1%8F%D0%B9%D1%82%D0%B5-%D0%B8%D0%B7%D0%BC%D0%B5%D0%BD%D0%B5%D0%BD%D0%B8%D0%B5-%D0%BE%D0%B1%D1%8A%D0%B5%D0%BA%D1%82%D0%BE%D0%B2-%D0%B8-%D1%81%D1%86%D0%B5%D0%BD%D0%B0%D1%80%D0%B8%D0%B8-%D1%83%D0%BC%D0%BD%D0%BE%D0%B3%D0%BE-%D0%B4%D0%BE%D0%BC%D0%B0